lms/djangoapps/courseware/views.py

```diff
@@ -894,6 +895,18 @@ def course_about(request, course_id):
             reg_then_add_to_cart_link = "{reg_url}?course_id={course_id}&enrollment_action=add_to_cart".format(
                 reg_url=reverse('register_user'), course_id=urllib.quote(str(course_id)))
 
+        ecommerce_checkout_link = ''
+        mode = ''
+        ecomm_service = EcommerceService()
+        if ecomm_service.is_enabled() and (
+            'professional' in modes or 'no-id-professional' in modes
```

Comment:
It drives me crazy to see these specific mode strings in the platform code -- I know they are everywhere, and I don't have a good solution at the moment, but there has to be a better way (enum?) to make the system more resilient.